app/api/models/domain/TaxYear.scala

```diff
@@ -119,23 +117,27 @@ object TaxYear {
   def fromDownstreamInt(taxYear: Int): TaxYear =
     new TaxYear(taxYear.toString)
 
-  type TodaySupplier = () => LocalDate
 
-  def currentTaxYear()(implicit todaySupplier: TodaySupplier = today _): TaxYear = {
-    val today            = todaySupplier()
-    val year             = today.getYear
-    val taxYearStartDate = LocalDate.parse(s"$year-04-06")
+  implicit val ordering: Ordering[TaxYear] = Ordering.by(_.year)
 
-    val taxYear =
-      if (today.isBefore(taxYearStartDate)) year
-      else year + 1
+  implicit val writes: Writes[TaxYear] = implicitly[Writes[String]].contramap(_.asMtd)
+}
 
-    new TaxYear(taxYear.toString)
-  }
+@ProvidedBy(classOf[TodaySupplier.type])
```

Comment:
There seem to be multiple solutions to the "date provider / today supplier" problem across the different APIs. If this one's definitely an improvement on the TodaySupplier in bsas-api, can it be updated there as well? (Or apply that one here). Either way, as long as they're consistent :)

Reply:
Yes and `TodaySupplier` is nothing to do with `TaxYear`. Something like a `Clock` might be simpler (and more standard).